OpenSim/Simulation/Model/ElasticFoundationForce.cpp

```diff
@@ -63,10 +63,17 @@ void ElasticFoundationForce::extendAddToSystem(SimTK::MultibodySystem& system) c
     {
         ContactParameters& params = contactParametersSet.get(i);
         for (int j = 0; j < params.getGeometry().size(); ++j) {
-            // get the ContactGeometry from the Model
-            const ContactGeometry& geom =
-                getModel().getComponent<ContactGeometry>(params.getGeometry()[j]);
-
+            // TODO: Dependency of ElasticFoundationForce on ContactGeometry
+            // should be handled by Sockets.
+            const ContactGeometry* contactGeom = nullptr;
+            if (getModel().hasComponent<ContactGeometry>(params.getGeometry()[j]))
+                contactGeom = &getModel().getComponent<ContactGeometry>(
+                    params.getGeometry()[j]);
+            else
+                contactGeom = &getModel().getComponent<ContactGeometry>(
+                    "./contactgeometryset/" + params.getGeometry()[j]);
+
+            const ContactGeometry& geom = *contactGeom;
             // B: base Frame (Body or Ground)
```

Comment:
If there was a typo or problem upstream wouldn't this blow up dereferencing nullptr?

Reply:
or did we guarantee earlier that contactGeom cannot be nullptr?

Reply:
The behavior is identical to before the PR. We do not dereference a nullptr because `getComponent()` throws if the component is not returned.